pkg/controller/endpoint/endpoints_controller.go

```diff
@@ -486,8 +486,9 @@ func (e *EndpointController) syncService(key string) error {
 		if errors.IsNotFound(err) {
 			currentEndpoints = &v1.Endpoints{
 				ObjectMeta: metav1.ObjectMeta{
-					Name:   service.Name,
-					Labels: service.Labels,
+					Name:        service.Name,
+					Labels:      service.Labels,
+					Annotations: service.Annotations,
```

Comment:
Sorry should have caught this earlier. I suspect this will not be safe. The last applied annotation (https://kubernetes.io/docs/concepts/overview/object-management-kubectl/declarative-config/#how-to-create-objects) is used to control writes from kubectl apply. I suspect that they will not be safe to blindly copy over. It would probably make sense to better understand the use case for copying annotations over. Maybe we can either filter (+/-) the annotations or there may be a better way to solve the use case. The new server side apply (https://github.com/kubernetes/community/blob/master/keps/sig-api-machinery/0006-apply.md) also involves annotations and would have problems with a blind copy.

Reply:
@cheftako any ideas on the list of annotations that should be "filtered-out"?

Reply:
I'm afraid I'm aware of a minimum list but not necessarily the complete list. I think it would be good to reach out to @matteosilv to better understand the use case before we proceed. We can also try to determine any other items which should be on the list. Some may be items which are safe but deemed to not belong. For instance service frequently indicates the load-balancer type under a cloud provider specific key. While this may not be harmful, I'm not sure that that annotation belongs on the endpoint. I did chat with the server side apply team and they mentioned they will actually be moving the apply state to its own metadata field. However that means we are at least a year away from the old apply field going away.